### PR TITLE
test: Cypress- js to ts migration tests for Dynamic Height feature

### PR DESCRIPTION
**Pull request by NandanAnantharamu**
PR is meant to improve Dynamic height tests and make them more readable

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Modal_Widget_spec.js [hunk 1]
```diff
@@ -1,39 +1,41 @@
...
+} from "../../../../support/Objects/ObjectsCore";
+
+const textMsg =
+  "Dynamic panel validation for text widget wrt heightDynamic panel validation for text widget wrt heightDynamic panel validation for text widget wrt height Dynamic panel validation for text widget Dynamic panel validation for text widget Dynamic panel validation for text widget";
 
 describe("Dynamic Height Width validation with limits", function () {
   it("1. Validate change in auto height with limits width for widgets and highlight section validation", function () {
-    const textMsg =
-      "Dynamic panel validation for text widget wrt heightDynamic panel validation for text widget wrt heightDynamic panel validation for text widget wrt height Dynamic panel validation for text widget Dynamic panel validation for text widget Dynamic panel validation for text widget";
     cy.addDsl(dsl);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Modal_Widget_spec.js:14]: Use TS method.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Modal_Widget_spec.js [hunk 2]
```diff
@@ -1,39 +1,41 @@
...
-                expect(mheight).to.not.equal(mnewheight);
-              });
-          });
-      });
+    entityExplorer.SelectEntityByName("Modal1", "Widgets");
+    agHelper.GetWidgetHeight(locators._modal).then((currentModalHeight) => {
+      entityExplorer.SelectEntityByName("Text1", "Modal1");
+      agHelper.AssertElementVisible(commonlocators.generalSectionHeight);
+      cy.changeLayoutHeightWithoutWait(commonlocators.autoHeight);
+      entityExplorer.SelectEntityByName("Text1", "Modal1");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Modal_Widget_spec.js:20]: Is this needed when its already selected in line 17?

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Modal_Widget_spec.js:20]: Can you brief what this test case is about?

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Modal_Widget_spec.js (hunk 1 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Modal_Widget_spec.js:14]: i will move the whole test to TS later , this is only meant to show u the handling of validating the height part alone

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Modal_Widget_spec.js (hunk 2 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Modal_Widget_spec.js:20]: let me check by removing that line, the test is about checking the auto height with modal and updating the text within modal. So Modal name is updated with the above text and changes of modal height is validated in both the cases one before auto height and after height and same applies to text within modal.

Isnt this more readable ? there are similar things in other usecases as well more 1 within another and other.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/CanvasHeight_resize_spec.js [hunk 3]
```diff
@@ -1,64 +1,83 @@
...
+          .GetWidgetHeight(locators._widgetInDeployed("textwidget"))
+          .then((initialTextWidgetHeight) => {
+            //Change the text label based on the textMsg above
+            cy.testCodeMirror(textMsg);
+            cy.moveToStyleTab();
+            cy.ChangeTextStyle(
+              this.data.TextHeading,
+              commonlocators.headingTextStyle,
+              textMsg,
+            );
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/CanvasHeight_resize_spec.js:31]: Can replace with TS methods once CY 12 is merged.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+              .then((currentContainerHeight: number) => {
+                expect(currentContainerHeight).to.not.equal(
+                  updatedContainerHeight,
+                );
+                agHelper
+                  .GetElement(locators._labelContains("Off"))
+                  .should("not.exist");
+                agHelper
+                  .GetElement(locators._labelContains("On"))
+                  .should("be.visible");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts:66]: aghelpers to be used.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        agHelper.Sleep(2000);
+        agHelper
+          .GetWidgetCSSHeight(locators._widgetInDeployed("containerwidget"))
+          .then((updatedContainerHeight: number) => {
+            expect(currentContainerHeight).to.not.equal(updatedContainerHeight);
+            agHelper
+              .GetElement(locators._labelContains("On"))
+              .should("not.exist");
+            agHelper
+              .GetElement(locators._labelContains("Off"))
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts:51]: All places like this.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            agHelper
+              .GetElement(locators._labelContains("On"))
+              .should("not.be.enabled");
+          });
+      });
+    deployMode.DeployApp();
+    agHelper
+      .GetWidgetCSSHeight(locators._widgetInDeployed("containerwidget"))
+      .then((currentContainerHeight: number) => {
+        agHelper.GetNClick(".bp3-control-indicator");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts:41]: NO direct locator.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Auto_Height_spec.ts [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+                )
+                .then((selectedFont) => {
+                  expect(dropdownFont).to.equal(selectedFont);
+                });
+            }
+          });
+      });
+    agHelper
+      .GetWidgetCSSFrAttribute(
+        `${locators._propertyControl}fontsize .rc-select`,
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Auto_Height_spec.ts:57]: No direct locator.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Auto_Height_spec.ts [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  it("1. Validate change with auto height width for widgets", function () {
+    cy.fixture("dynamicHeightContainerCheckboxdsl").then((val) => {
+      agHelper.AddDsl(val);
+    });
+    entityExplorer.SelectEntityByName("Container1", "Widgets");
+    entityExplorer.SelectEntityByName("CheckboxGroup1", "Container1");
+    propPane.MoveToTab("Style");
+    agHelper
+      .GetWidgetCSSFrAttribute(
+        `${locators._propertyControl}fontsize .rc-select`,
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Auto_Height_spec.ts:27]: No direct locator,

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+          .GetWidgetCSSHeight(locators._widgetInDeployed("containerwidget"))
+          .then((updatedContainerHeight: number) => {
+            expect(currentContainerHeight).to.not.equal(updatedContainerHeight);
+            agHelper
+              .GetElement(locators._labelContains("On"))
+              .should("not.exist");
+            agHelper
+              .GetElement(locators._labelContains("Off"))
+              .should("be.visible");
+            agHelper.GetNClick(".bp3-control-indicator");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts:53]: No direct locator. Please check all changes again for direct locators.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Form_With_SwitchGroup_spec.ts [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+                      {"label": "Pink","value": "PINK"},
+                      {"label": "Black","value": "BLACK"},
+                      {"label": "Grey","value": "GREY"},
+                      {"label": "Orange","value": "ORANGE"},
+                      {"label": "Cream","value": "CREAM"}
+                    ]`,
+                );
+                agHelper
+                  .GetWidgetCSSHeight(
+                    locators._widgetInDeployed("switchgroupwidget"),
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Form_With_SwitchGroup_spec.ts:43]: Use widgets from draggableWidgets. No direct hardcoding like this - `switchgroupwidget`

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Form_With_SwitchGroup_spec.ts [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+                        expect(CurrentSwitchHeight).to.not.equal(
+                          UpdatedSwitchHeight,
+                        );
+                      });
+                  });
+              });
+          });
+      });
+    agHelper.GetNClick(
+      ".t--draggable-switchgroupwidget .bp3-control-indicator",
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Form_With_SwitchGroup_spec.ts:63]: No direct locator.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Form_With_SwitchGroup_spec.ts [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+    );
+    agHelper.Sleep(3000);
+    agHelper.GetElement(locators._modal).should("have.length", 1);
+    agHelper
+      .GetElement(".t--widget-propertypane-toggle")
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Form_With_SwitchGroup_spec.ts:68]: No direct locator.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/List_Resizing_spec.ts [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+describe("Dynamic Height Width validation", function () {
+  it("1. Validate change with auto height width for List widgets", function () {
+    cy.fixture("ResizeListDsl").then((val) => {
+      agHelper.AddDsl(val);
+    });
+    entityExplorer.SelectEntityByName("Tab 1", "Tabs1");
+    entityExplorer.SelectEntityByName("List1", "Tab 1");
+    agHelper
+      .GetWidgetCSSHeight(locators._widgetInDeployed("listwidgetv2"))
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/List_Resizing_spec.ts:18]: No hardcodings - `listwidgetv2`

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/List_TextWidget_Spec.ts [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+        assertHelper.AssertNetworkStatus("@updateLayout");
+        agHelper.Sleep(2000);
+        entityExplorer.NavigateToSwitcher("Explorer");
+        entityExplorer.SelectEntityByName("Text3Copy");
+        agHelper.AssertElementAbsence(locators._propertyPaneHeightLabel);
+        agHelper.TypeText(locators._body, `{${modifierKey}}c`, 0, true);
+
+        agHelper
+          .GetElement("[data-testid='div-selection-0']")
+          .click({ force: true });
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/List_TextWidget_Spec.ts:76]: No direct locator. Why is GetNClick not used here? any reason?

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/List_TextWidget_Spec.ts [hunk 15]
```diff
@@ -0,0 +1,130 @@
+import {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/List_TextWidget_Spec.ts:1]: Too many hard coded waits added in this spec. All needed?

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/List_spec.ts [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+    });
+    entityExplorer.SelectEntityByName("List1");
+    agHelper
+      .GetWidgetCSSHeight(locators._widgetInDeployed("listwidget"))
+      .then((currentListHeight: number) => {
+        agHelper.AssertElementAbsence(locators._propertyPaneHeightLabel);
+        entityExplorer.SelectEntityByName("Container1", "List1");
+        entityExplorer.SelectEntityByName("Text1", "Container1");
+        agHelper.AssertElementAbsence(locators._propertyPaneHeightLabel);
+        propPane.TypeTextIntoField("text", textMsg, true);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/List_spec.ts:24]: Recommendation was to use UpdatePropertyField() right? Any reason for TypeText here & all places used??

File: app/client/cypress/support/Objects/CommonLocators.ts [hunk 17]
```diff
@@ -225,6 +225,24 @@ export class CommonLocators {
...
+  _autoHeightLimitMin_div =
+    "[data-testid='t--auto-height-overlay-handles-min'] div";
+  _autoHeightLimitMax = "[data-testid='t--auto-height-overlay-handles-max']";
+  _labelContains = (value: string) => `label:Contains('${value}')`;
+  _showColumnButton = ".t--show-column-btn";
+  _toastBody = ".Toastify__toast-body span";
+  _propertyPaneHeightLabel =
+    ".t--property-pane-section-general .t--property-control-label:contains('Height')";
+  _listCollapseToggle =
+    ".t--entity-item:contains('List1') .t--entity-collapse-toggle";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Objects/CommonLocators.ts:239]: Why is List1 hardcoded? Get it from caller.

File: app/client/cypress/support/Objects/CommonLocators.ts [hunk 18]
```diff
@@ -225,6 +225,24 @@ export class CommonLocators {
   _selectionCanvas = (canvasId: string) => `#div-selection-${canvasId}`;
   _sqlKeyword = ".cm-m-sql.cm-keyword";
   _appLeveltooltip = (toolTip: string) => `span:contains('${toolTip}')`;
+  _submitBtn = "span:contains('Submit')";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Objects/CommonLocators.ts:228]: Use ClickButton() for Submit. Remove this locator.

File: app/client/cypress/support/Objects/CommonLocators.ts [hunk 19]
```diff
@@ -225,6 +225,24 @@ export class CommonLocators {
...
+  _autoHeightLimitMin = "[data-testid='t--auto-height-overlay-handles-min']";
+  _autoHeightLimitMin_div =
+    "[data-testid='t--auto-height-overlay-handles-min'] div";
+  _autoHeightLimitMax = "[data-testid='t--auto-height-overlay-handles-max']";
+  _labelContains = (value: string) => `label:Contains('${value}')`;
+  _showColumnButton = ".t--show-column-btn";
+  _toastBody = ".Toastify__toast-body span";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Objects/CommonLocators.ts:235]: Toast locators are already present. Can they be used?

File: app/client/cypress/support/Pages/AssertHelper.ts [hunk 20]
```diff
@@ -82,6 +82,12 @@ export class AssertHelper extends ReusableHelper {
     // });
   }
 
+   public WaitForNetworkCall(aliasName: string, expectedStatus = 200) {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/AssertHelper.ts:85]: AssertNetworkStatus - already waits & validates it, Why is this needed?

File: app/client/cypress/support/Pages/AggregateHelper.ts [hunk 21]
```diff
@@ -604,6 +604,20 @@ export class AggregateHelper extends ReusableHelper {
       .wait(waitTimeInterval);
   }
 
+  public GetClosestNClick(
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/AggregateHelper.ts:607]: Remove this unused method.

Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/AggregateHelper.ts:622]: This we will be using for AppNavigation tests hence added.

File: app/client/cypress/support/Pages/AssertHelper.ts (hunk 20 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/AssertHelper.ts:85]: The existing method is used for validation this was added to validate the response data for further testing

File: app/client/cypress/support/Objects/CommonLocators.ts (hunk 19 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Objects/CommonLocators.ts]: done replaced with existing one

File: app/client/cypress/support/Objects/CommonLocators.ts (hunk 18 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Objects/CommonLocators.ts:228]: our usecase is slightly different hence ClickButton was not used.

File: app/client/cypress/support/Objects/CommonLocators.ts (hunk 17 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Objects/CommonLocators.ts]: done removed

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Auto_Height_spec.ts (hunk 7 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Auto_Height_spec.ts]: done removed

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts (hunk 6 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts]: done

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts (hunk 5 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts]: done

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts (hunk 4 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/DynamicHeight_Visibility_spec.ts]: done

File: app/client/cypress/support/Pages/AggregateHelper.ts [hunk 22]
```diff
@@ -1348,8 +1362,16 @@ export class AggregateHelper extends ReusableHelper {
     });
   }
 
-  GetWidgetCSSHeight(widgetSelector: string) {
-    return this.GetElement(widgetSelector).invoke("css", "height");
+  public GetWidgetCSSHeight(widgetSelector: string, index = 0) {
+    return this.GetElement(widgetSelector).eq(index).invoke("css", "height");
+  }
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/AggregateHelper.ts:1371]: Let this method be last. Please dont create anymore duplicates for methods already present.

File: app/client/cypress/support/Objects/CommonLocators.ts (hunk 18 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/support/Objects/CommonLocators.ts:243]: I dont see this locator usage. Where is it used? Would like to understand the usage.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/List_spec.ts (hunk 16 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/List_spec.ts:23]: ?

File: app/client/cypress/e2e/Regression/ClientSide/AppNavigation/Sidebar_spec.ts [hunk 23]
```diff
@@ -0,0 +1,111 @@
...
+  locators,
+} from "../../../../support/Objects/ObjectsCore";
+
+describe("Test Sidebar navigation style", function () {
+  before(() => {
+    // Import an application
+    homePage.NavigateToHome();
+    homePage.ImportApp("appNavigationTestingAppWithLongPageNamesAndTitle.json");
+
+    cy.wait("@importNewApplication").then((interception) => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/AppNavigation/Sidebar_spec.ts:15]: New method not used here?

File: app/client/cypress/e2e/Regression/ClientSide/AppNavigation/TopInline_spec.ts [hunk 24]
```diff
@@ -0,0 +1,154 @@
...
+      .parent()
+      .parent()
+      .should("have.class", "is-active");
+  });
+
+  it("5. Navigation's background should be default to white, and should change when background color is set to theme", () => {
+    // The background should be white since light color style is default
+    agHelper
+      .GetElement(appSettings.locators._header)
+      .should("have.css", "background-color", "rgb(255, 255, 255)");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/AppNavigation/TopInline_spec.ts:114]: Can `agHelper.AssertCSS` be used here & all places?

File: app/client/cypress/support/Objects/CommonLocators.ts (hunk 18 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/support/Objects/CommonLocators.ts:243]: Add all new locators only to last.

**Review by Aishwarya-U-R — COMMENTED** [review]
Please dont add any more files to this PR. It exhausting to review long files & few things can easily slip thru in such cases. I would also recommend moving AppNavigation out.

File: app/client/cypress/support/Pages/AggregateHelper.ts (hunk 22 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/AggregateHelper.ts:1371]: ok

**Comment by NandanAnantharamu** [comment]
https://github.com/appsmithorg/appsmith/pull/24328#discussion_r1250433231 
Here if u see Inside List its the container which has image and text widget, therefore to access the text widget inside List we need to use List > Container> Text hence used it.

File: app/client/cypress/e2e/Regression/ClientSide/AppNavigation/TopInline_spec.ts (hunk 24 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/AppNavigation/TopInline_spec.ts:114]: ok

**Comment by Aishwarya-U-R** [comment]
> TypeTextIntoField

No here review comment is about usage of TypeTextIntoField as opposed to UpdatePropertyField that was not addressed.

**Comment by Aishwarya-U-R** [comment]
@NandanAnantharamu - [DynamicHeight/List_TextWidget_Spec.ts](https://cloud.cypress.io/projects/eyxvp8/runs/37006/test-results?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%7B%22value%22%3Atrue%2C%22label%22%3A%22Flaky%22%7D%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%7B%22label%22%3A%22Regression%2FClientSide%2FDynamicHeight%2FList_TextWidget_Spec.ts%22%2C%22value%22%3A%22%5B%5C%224b3f2120-780d-4053-8851-dd8483ae6c14%5C%22%5D%22%7D%5D&statuses=%5B%5D&testingTypesEnum=%5B%5D) is flaky. Please fix it.

**Comment by Aishwarya-U-R** [comment]
Cypress tests in TS looks ok. Except for flaky fix & locators movement.

**Comment by Aishwarya-U-R** [comment]
https://github.com/appsmithorg/appsmith/pull/24328#discussion_r1250426027 not addressed yet.

File: app/client/cypress/support/Pages/AppSettings/AppSettings.ts [hunk 25]
```diff
@@ -27,6 +27,39 @@ export class AppSettings {
     _getPageSettingsHeader: (pageName: string) =>
       `#t--page-settings-${pageName}`,
     _updateStatus: ".ads-v2-icon.rotate",
+    _header: ".t--app-viewer-navigation-header",
+    _topStacked: ".t--app-viewer-navigation-top-stacked",
+    _applicationName: ".t--app-viewer-application-name",
+    _shareButton: ".t--app-viewer-share-button",
+    _editButton: ".t--back-to-editor",
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/AppSettings/AppSettings.ts:34]: This locator `_editButton` is already present. Please remove the duplicate. Check for the locator & then add.

Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/AppSettings/AppSettings.ts:34]: As part of review - its difficult to check all locators if a duplicate is present. So double check from your end, before adding any new locators.

Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/AppSettings/AppSettings.ts:34]: Ok, so all of these are duplicates from AppNavigation.json. Even while adding like this into ts - double check!

File: app/client/cypress/support/Pages/AssertHelper.ts [hunk 26]
```diff
@@ -90,6 +90,12 @@ export class AssertHelper extends ReusableHelper {
     // });
   }
 
+  public WaitForNetworkCall(aliasName: string, expectedStatus = 200) {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/AssertHelper.ts:93]: Is latest over-ridden? I have added this method. Use that.

Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/AssertHelper.ts:93]: Remove this method. I see my definition of WaitForNetworkCall() also present. Weird how this did not throw error in this branch or is unnoticed.

File: app/client/cypress/support/Pages/AppSettings/AppSettings.ts (hunk 25 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/AppSettings/AppSettings.ts:34]: this cleanup i will do when i am sending for Appnavigation its on me

Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/AppSettings/AppSettings.ts:34]: No dint get, this PR will go before AppNavigation PR right?

File: app/client/cypress/support/Pages/AssertHelper.ts (hunk 26 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/AssertHelper.ts]: done

File: app/client/cypress/support/Pages/AppSettings/AppSettings.ts (hunk 25 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/AppSettings/AppSettings.ts:34]: yes .. this will go before AppNavigation, i will correct with that PR by removing duplicate ones

Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/AppSettings/AppSettings.ts:34]: If a PR is already merged I understand doing it in next PR. Why is this not corrected in this PR is still unclear to me, @NandanAnantharamu - please note this will not be encouraged going forward.

File: app/client/cypress/support/Pages/AggregateHelper.ts (hunk 21 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/AggregateHelper.ts:623]: Let things specific to PR go in that PR. These things will not be merged next time.

File: app/client/cypress/support/Objects/CommonLocators.ts [hunk 27]
```diff
@@ -250,4 +250,11 @@ export class CommonLocators {
   _gitStatusChanges = "[data-testid='t--git-change-statuses']";
   _appNavigationSettings = "#t--navigation-settings-header";
   _appNavigationSettingsShowTitle = "#t--navigation-settings-application-title";
+  _switchGroupControl =
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Objects/CommonLocators.ts:253]: Un-used locator. Remove in next PR which will fix other things.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/List_Resizing_spec.ts [hunk 28]
```diff
@@ -0,0 +1,30 @@
...
+      .then((currentListHeight: number) => {
+        propPane.MoveToTab("Style");
+        propPane.UpdatePropertyFieldValue("Item Spacing (px)", "16");
+        agHelper
+          .GetWidgetCSSHeight(
+            locators._widgetInDeployed(draggableWidgets.LIST_V2),
+          )
+          .then((updatedListHeight: number) => {
+            expect(currentListHeight).to.equal(updatedListHeight);
+            agHelper.GetNAssertContains(locators._pagination, "5");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/List_Resizing_spec.ts:26]: Use `table.AssertPageNumber_List`

File: app/client/cypress/support/Objects/CommonLocators.ts [hunk 29]
```diff
@@ -250,4 +250,11 @@ export class CommonLocators {
...
+    ".t--draggable-switchgroupwidget .bp3-control-indicator";
+  _fontSelect = "fontsize .rc-select";
+  _fontInput = "fontsize input";
+  _pagination = ".rc-pagination";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Objects/CommonLocators.ts:257]: Remove this locator also.

File: app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Tab_spec.ts [hunk 30]
```diff
@@ -0,0 +1,98 @@
...
+                agHelper.AssertElementAbsence(propPane._tabId1);
+                agHelper.AssertElementAbsence(propPane._tabId2);
+              });
+          });
+      });
+    //it("Tab widget validation of height with reload", function() {
+    entityExplorer.SelectEntityByName("Tabs1");
+    agHelper.AssertElementVisible(propPane._propertyPaneHeightLabel);
+    agHelper.GetNClick(propPane._showTabsProperty);
+    propPane.SelectPropertiesDropDown("height", "Auto Height");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/DynamicHeight/Tab_spec.ts:82]: If this method `SelectPropertiesDropDown` is working means - Height dropdown & label is present. Line # 80 is redundant. Remove this & all such things.

**Review by Aishwarya-U-R — APPROVED** [review]
@NandanAnantharamu - Approving this PR now, since there was enough delay on this PR already. But all review comments **should** be addressed in next PR.
